Rewrite DayTimeMoneyBox and TimeOfDayString patches

Enerrex opened the pull request:
Rewrites the patches for the timeOfDayString function and the DayTimeMoneyBox draw function. 

Reorganize how the patches are initialized to reduce the visual burden in the mod entry. Made access to the logger static as the initializers were just passing in the original anyway.

The DayTimeMoneyBox draw patch was completely rewritten, while taking notes from the original. Instead of a prefix patch, we run the patch as a postfix. The original draw function will run, handle any of the relevant timer updates and determine the date text. We will then run the patch, rerender the backdrop UI, and render out our updated time string while rerendering the weather, season, time dial, and date text. Doing this may seem like somewhat of a waste, but it allows us to fine to the control of the box in a more consistent, organized, and easy to read manner while also leaving the money and quest displays untouched (and avoiding having to duplicate that logic as well).

This should allow the quest progress popup to display with the mod activated, and hopefully will fixup any issues with the time or date strings displaying incorrectly (though may have also introduced bugs in the latter, will solicit feedback!)

I don't think anybody is going to be reading this but I'm recording it for posterity.

Jpritch71 committed:
Rewrite the patches involved in displaying time as a string and in the DayTimeMoneyBox to fix issues hiding quest progress popups, and hopefully address a few other issues, while also cleaning up the code

KariArisu commented:
I'm not sure if this was still a work-in-progress or not, but I've been trying it out with these changes and the only noticeable bug I've ran into was when trying to enter buildings that aren't open yet. Instead of saying they are open from "9:00 am to 4:00 pm" it will say, for example, "9:03 am to 4:03 pm." It functions normally and opens/closes at the correct times, but it adds whatever the current minute is to end of the times in dialogue.

Enerrex commented:
> I'm not sure if this was still a work-in-progress or not, but I've been trying it out with these changes and the only noticeable bug I've ran into was when trying to enter buildings that aren't open yet. Instead of saying they are open from "9:00 am to 4:00 pm" it will say, for example, "9:03 am to 4:03 pm." It functions normally and opens/closes at the correct times, but it adds whatever the current minute is to end of the times in dialogue.

Thanks for testing it out, it is still a WIP but that bug is fixed in the code I have locally. I just haven't been very diligent about pushing code up. Do let me know if you encounter anything else!

Jpritch71 committed:
Remove TimeOfDayStringPatches prefix causing fixed time displays to erroneously display time with minutes added on.
Move pause time X draw to the last step so it is always over every piece of the box.
Add a silly comment.

Jpritch71 committed:
Minor formatting
